### PR TITLE
Swap 'base58' with 'bs58'

### PR DESCRIPTION
**Pull request by davxy**
Motivations:
- `base58` has hard limit of 128 bytes output (`bs58` doesn't have any limit)
   - with the introduction of https://github.com/paritytech/substrate/pull/13618 we require to decode longer strings
- (marginal) `bs58` is also used by `libp2p` so we depend on it anyway
- (marginal) `bs58` docs declares to be faster but I found the two almost equivalent.

### CODE REVIEW

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: cargo-check-each-crate-macos
**Logs**: https://gitlab.parity.io/parity/mirrors/substrate/-/jobs/2603631

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: cargo-check-each-crate-macos
**Logs**: https://gitlab.parity.io/parity/mirrors/substrate/-/jobs/2603681

File: primitives/core/Cargo.toml [hunk 1]
```diff
@@ -25,7 +22,7 @@ primitive-types = { version = "0.12.0", default-features = false, features = ["c
 impl-serde = { version = "0.4.0", optional = true }
 hash-db = { version = "0.16.0", default-features = false }
 hash256-std-hasher = { version = "0.15.2", default-features = false }
-base58 = { version = "0.2.0", optional = true }
+bs58 = { version = "0.4.0", optional = true }
```
Comment (koute) [inline primitives/core/Cargo.toml:25]: Shouldn't we add `default-features = false` to disable `bs58`'s `std` feature and add it below in `#[features]`?

Comment (bkchr) [reply primitives/core/Cargo.toml:25]: It is `no_std` by default. Their `std` feature is actually just `alloc`

**Review by bkchr — APPROVED** [review]

File: primitives/core/Cargo.toml (hunk 1 above)
Comment (koute) [reply primitives/core/Cargo.toml:25]: > It is `no_std` by default. Their `std` feature is actually just `alloc`

From what I can see in the code it isn't? Unless I'm misunderstanding something.

e.g. from https://github.com/Nullus157/bs58-rs/blob/prīmum/src/lib.rs

```rust
#[cfg(feature = "std")]
extern crate std;
```

and then in https://github.com/Nullus157/bs58-rs/blob/prīmum/src/encode.rs

```rust
#[cfg(feature = "std")]
impl std::error::Error for Error {}
```

Comment (bkchr) [reply primitives/core/Cargo.toml:25]: Okay weird, I had only checked the `lib.rs` file :shrug: https://github.com/Nullus157/bs58-rs/blob/pr%C4%ABmum/src/lib.rs#L1

**Pull request merged**